File: solutions/ruby/collatz-conjecture/1/collatz_conjecture.rb
class CollatzConjecture
  def self.steps(number)
    raise ArgumentError if number < 1

    stepCount = 0
    currentValue = number

    while currentValue > 1
      if currentValue.even?
        currentValue /= 2
      else
        currentValue = 3 * currentValue + 1
      end

      stepCount += 1
    end

    stepCount
  end
end

File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int number)
{
    if (number <= 0)
        return ERROR_VALUE;

    int count = 0; // Keep in scope after the loop.

    for ( ; number != 1; count++)
    {
        if ((number & 1) == 1)
        {
            number *= 3;
            number++;
            count++;
        }

        number = number >> 1;
    }

    return count;
}
